url9ef4af9ae5_stevenreyn_RegionViewer_tgz-pJ8-net_slreynolds_ds_SecondTestJ8/src/net/slreynolds/ds/internal/util/FileUtil.java

```diff
@@ -0,0 +1,35 @@
+
+package net.slreynolds.ds.internal.util;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Some utility methods on files
+ */
+public class FileUtil {
+    
+    public static File createEmptyWritableFile(String path) throws IOException {
+        File out = new File(path);
+        if (out.exists()) {
+            if (!out.isFile()) {
+                throw new IOException(path + " is not a file");
+            }
+            if (!out.delete()) {
+                throw new IOException("Unable to delete existing file: "+path);
+            }
+        }
+        // invariant: file does not exist 
+        File parent = out.getParentFile();
+        if (parent != null && !parent.exists()) {
+            parent.mkdirs();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Confirm that `mkdirs` was successful. Handle any failures that happened. If you fail silently, then you might encounter errors that are harder to debug. See the following diffs on GitHub that make this code change: [diff1](https://github.com/qunarcorp/qmq/commit/12f1d9d6a57a02bc2415cb2af9f4388a77b12c02#diff-6b67be6e17f24fdeddb7c725ac8ec5600d6e63be0bdca0963a83d530e6d55e35L36), [diff2](https://github.com/apache/jmeter/commit/1ec61206779b3f84df18516682b4791bf015178f#diff-cd74304763c097fbc085ac2bc952b46ab3843ac825cacacad89e8064296e955aL330)

Learn more: [File.mkdirs in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#mkdirs--)